openziti-init-entrypoint.sh

```diff
@@ -0,0 +1,80 @@
+  ziti edge delete service "${svc}"
+  ziti edge delete erp "${svc}.erp"
+  ziti edge delete sp "${svc}.bind"
+  ziti edge delete sp "${svc}.dial"
+}
+
+function makeEdgexService {
+  local svc="edgex.$1"
+  local int="$1.edgex.ziti"
+  ziti edge create identity "${svc}" --external-id "${1}" -a "edgex,${svc}-servers" -P "${edgex_auth_policy}"
+  ziti edge create config "${svc}-intercept" intercept.v1 '{"protocols":["tcp"],"addresses":["'"${int}"'"], "portRanges":[{"low":80, "high":80}]}'
+  ziti edge create service "${svc}" -a "${svc}" --configs "${svc}-intercept"
+  ziti edge create edge-router-policy "${svc}.erp" --identity-roles "#${svc}" --edge-router-roles '#all'
+  ziti edge create service-policy "${svc}.bind" Bind --service-roles "#${svc}" --identity-roles "#${svc}-servers"
+  ziti edge create service-policy "${svc}.dial" Dial --service-roles "#${svc}" --identity-roles "#${svc}-clients"
+}
+
+sleep 1
+makeEdgexService 'core-command'
+makeEdgexService 'core-data'
+makeEdgexService 'core-metadata'
+makeEdgexService 'device-virtual'
+makeEdgexService 'device-rest'
+makeEdgexService 'rules-engine'
+makeEdgexService 'support-notifications'
+makeEdgexService 'support-scheduler'
+makeEdgexService 'sys-mgmt-agent'
+makeEdgexService 'ui'
+makeEdgexService 'app-rules-engine'
+
+
+ziti edge create identity edgexuser -P "${edgex_auth_policy}" --external-id edgexuser -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.rules-engine -a edgex.rules-engine-servers,edgex.core-command-clients
+ziti edge update identity edgex.device-virtual -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.ui -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.device-rest -a edgex.device-rest-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.core-command -a edgex,edgex.core-command-servers,edgex.core-metadata-clients,edgex.device-virtual-clients
+
+echo "granting additional dial capabilities to app-rules-engine"
+ziti edge update identity edgex.app-rules-engine -a edgex,edgex.app-rules-engine-servers,edgex.core-metadata-clients
```

Comment:
Will this be needed for all app services?
Move this to `makeEdgexService`, add 2nd parameter for service type (`core` `support`, `app` device`) and only do this if `type=app`

Reply:
it does all kind of "depend". Will all app-services receive a jwt from Vault they can use to authenticate to the overlay with? We can certainly make it easier by creating a "any edgex identity can communicate to any edgex identity" scheme if that's desired. This might end up needing to be a live conversation.

Reply:
Yes, all App Service are just like App Rules Engine in this respect. Same with all Device services

Reply:
OpenZiti has attributes we can use. I can refactor these permsisions to use attributes like:
* core-services
* app-services
* devices-services

Any app-service or device-service would be authorized to communicate to any core-service if that's what you'd like to see. Should all app-services be authorized to commuicated to all app-services, all device-services? That is certainly easier mentally, but does reduce the security posture somewhat?

Or doe all services have the authorization to communicate to core-metadata? That's the only thing app-rules-engine is allowed to communicate to per this policy.

Reply:
See https://github.com/edgexfoundry/edgex-compose/pull/440#discussion_r1538276301